lib/data/models/user/user_provider.dart

```diff
@@ -1,24 +1,22 @@
 import 'package:flutter/material.dart';
 import 'package:graphql_flutter/graphql_flutter.dart';
 import 'package:mm_flutter_app/data/models/base/base_provider.dart';
-import 'package:mm_flutter_app/data/models/user/queries/find_users.dart';
-import 'package:mm_flutter_app/data/models/user/queries/get_authenticated_user.dart';
-import 'package:mm_flutter_app/data/models/user/queries/get_user_profile_info.dart';
-import 'package:mm_flutter_app/data/models/user/user.dart';
-import 'package:mm_flutter_app/data/models/user/user_api.dart';
+import 'package:mm_flutter_app/utilities/utility.dart';
 import 'package:shared_preferences/shared_preferences.dart';
-import 'package:uuid/uuid.dart';
 
-class UserProvider extends BaseProvider {
-  GraphQLClient client;
+import '../../../__generated/schema/operations.graphql.dart';
+import '../../../__generated/schema/schema.graphql.dart';
+import '../base/operation_result.dart';
 
-  User? _user;
+class UserProvider extends BaseProvider {
+  Query$GetAuthenticatedUser$getAuthenticatedUser? _user;
```

Comment:
From what I understand Fragments are defined in the schema, which is in the backend (not in our scope). I agree that these are very long, but I see why they went with something like that. We could just use a dynamic variable whenever the type can be easily deducted from the usage (widgets and other classes that call this code).

Reply:
Agreed with this, it would be nice for the Flutter representation of a user variable to look more like `final User user` as it does currently, rather than `final Query$GetAuthenticatedUser$getAuthenticatedUser user`. I don't know how to reconcile this with the changes in this PR, though, so I don't have any suggestion for that.

Reply:
I could provide an alias for this and other types that get passed around a lot. How about AuthenticatedUser? I just don't want to obscure the fact that this is not a User as defined in the schema, and that it is instead the result of the getAuthenticatedUser query.

Reply:
That looks great to me. Thanks Miguel!